src/main/java/de/medizininformatikinitiative/flare/model/fhir/Resource.java

```diff
@@ -3,9 +3,13 @@
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 
 @JsonIgnoreProperties(ignoreUnknown = true)
-public record Resource(String id, Reference patient, Reference subject) {
+public record Resource(String id, Reference patient, Reference subject, String resourceType) {
 
     public String patientId() {
         return patient != null ? patient.id() : subject != null ? subject.id() : id;
     }
+
+    public boolean hasReference(){
```

Comment:
I would remove that method and use the optional result of `patientId()` instead.